jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -173,6 +198,44 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
     def vue_spectral_extraction(self, *args, **kwargs):
         self.collapse_to_spectrum(add_data=True)
 
+    def vue_save_as_fits(self, *args):
+        self._save_extracted_spec_to_fits()
+
+    def _save_extracted_spec_to_fits(self, overwrite=False, *args):
+
+        if not self.export_enabled:
+            # this should never be triggered since this is intended for UI-disabling and the
+            # UI section is hidden, but would prevent any JS-hacking
+            raise ValueError("Writing out extracted spectrum to file is currently disabled")
+
+        # Make sure file does not end up in weird places in standalone mode.
+        path = os.path.dirname(self.filename)
+        if path and not os.path.exists(path):
+            raise ValueError(f"Invalid path={path}")
+        elif (not path or path.startswith("..")) and os.environ.get("JDAVIZ_START_DIR", ""):  # noqa: E501 # pragma: no cover
+            filename = Path(os.environ["JDAVIZ_START_DIR"]) / self.filename
+        else:
+            filename = Path(self.filename).resolve()
+
+        if filename.exists():
+            if overwrite:
+                # Try to delete the file
+                filename.unlink()
+                if filename.exists():
+                    # Warn the user if the file still exists
+                    raise FileExistsError(f"Unable to delete {filename}. Check user permissions.")
+            else:
+                self.overwrite_warn = True
+                return
```

Comment:
When I try to save an extracted spectrum with the same name in the same location as a previous save, I see no notification that a file with that name already exists.

Reply:
Good catch!  Moment maps has an overlay that gets triggered in this case and requires the user confirming the overwrite.  Without that and having the default from `vue_save_as_fits`, this looks like it just silently ignores when the file already exists and will not overwrite it.